src/usdb_syncer/song_loader.py

```diff
@@ -577,8 +579,15 @@ def _write_mp3_tags(
     tags.save(path)
 
 
-def _write_ogg_tags(path: Path, ctx: _Context, embed_artwork: bool) -> None:
-    audio = mutagen.oggvorbis.OggVorbis(path)
+def _write_ogg_tags(
+    path: Path, resource: str, ctx: _Context, embed_artwork: bool
+) -> None:
+    audio: mutagen.ogg.OggFileType
+    match path.suffix:
+        case ".ogg":
+            audio = mutagen.oggvorbis.OggVorbis(path)
+        case ".opus":
+            audio = mutagen.oggopus.OggOpus(path)
```

Comment:
If `path.suffix` is anything else, `audio` will be unbound and the function will throw. This is problematic because its signature doesn't state this requirement. pylint only accepts it due to the explicit annotation in line 585.
You could change the `path: Path` parameter to `audio: mutagen.ogg.OggFileType` and handle the ogg/opus distinction in the match statement above.

Reply:
Thanks, good point! You mean like this?

Reply:
Yes, exactly. And thanks for correcting my typo. Should probably also be `_cleanup_existing_resourceS`, but whatever. 😅